docs/guides/database.rst

```diff
@@ -193,9 +215,21 @@ $type
 $subtype
 	Entity subtype, e.g. ``'blog'`` or ``'page'``.
 
-By convention entities that have an uploaded avatar or icon will have the ``icontime`` property
-assigned. This means that you can use ``$entity->icontime`` to check if an icon exists for the given
-entity.
+Icons are assigned a timestamp when they're created. You can use ``$object->getIconTs($size)`` to 
+get the timestamp if you need to track changes to the icon.
+
+To check if an icon is set, use ``$object->hasIcon($size)``.
+
+Icon methods support passing an icon type if an entity has more than one icon. For example, a user
+might have an avatar and a cover photo icon. You would pass ``'cover_photo'`` as the icon type:
+
+.. code:: php
+	$object->setIconFromUploadedFile('uploaded_photo', 'cover_photo');
+
+	$object->getIconUrl([
+		'size' => 'medium',
+		'type' => 'cover_photo'
```

Comment:
> I don't think these are exclusive goals. The problem I'm trying to solve is separating visual design concerns (e.g., this image is 200x300 pixels) from implementation concerns across plugins (display a "large" icon), while allowing plugins to redefine "large" to suit their needs, which includes regeneration and caching of icons on the fly without exposing the DOS attack I described above.

Ok. I missed the "on the fly" part. That changes things, as I don't think we can have both - engine-less icon serving and on the fly icon generation.